File: gdx2d-desktop/src/hevs/gdx2d/demos/simple/mistfy/DemoLines.java
package hevs.gdx2d.demos.simple.mistfy;

import hevs.gdx2d.lib.GdxGraphics;
import hevs.gdx2d.lib.PortableApplication;

/**
 * A classical Mistify screen saver clone. 
 * 
 * Code adapted from http://r3dux.org/2010/11/mystify-2-0/
 * @author Pierre-André Mudry
 * @version 1.0
 * 
 */
public class DemoLines extends PortableApplication {
	BounceShape[] s;
	
	final int N_SHAPES = 3;
	int frame = 0;
	
	public DemoLines(boolean onAndroid) {
		super(onAndroid);
	}

	@Override
	public void onInit() {
		// Sets the window title
		setTitle("Mistfy screensaver clone, mui 2013");

		// Allocate size for the shapes
		s = new BounceShape[N_SHAPES];
		
		// Initialize them
		for(int i = 0; i < N_SHAPES; i++){
			s[i] = new BounceShape(this.getWindowWidth(), this.getWindowHeight());
		}
	}

	@Override
	public void onGraphicRender(GdxGraphics g) {		
		
		// Clears the screen
		g.clear();

		// Draws and moves the shapes
		for(BounceShape shape : s){
			shape.drawShape(g);
			shape.moveShape(g.getScreenWidth(), g.getScreenHeight());
			shape.shiftShapeColour(frame++);
		}
		
		g.drawFPS();
		g.drawSchoolLogo();
	}
	
	public static void main(String[] args) {
		/**
		 * Note that the constructor parameter is used to determine if running
		 * on Android or not. As we are in main there, it means we are on
		 * desktop computer.
		 */
		new DemoLines(false);
	}

}

File: gdx2d-demoDesktop/src/main/scala/ch/hevs/gdx2d/demos/simple/mistify/DemoLines.scala
package ch.hevs.gdx2d.demos.simple.mistify

import ch.hevs.gdx2d.desktop.DesktopApplication
import ch.hevs.gdx2d.lib.GdxGraphics

/**
 * A classical Mistify screen saver clone.
 *
 * Code adapted from http://r3dux.org/2010/11/mystify-2-0/
 *
 * @author Pierre-André Mudry
 * @version 2.0
 */
class DemoLines extends DesktopApplication {

  private val N_SHAPES = 3
  private var s: Array[BounceShape] = _
  private var frame: Int = 0

  override def onInit(): Unit = {
    setTitle("Mistify screensaver clone, mui 2013")
    s = Array.fill(N_SHAPES)(new BounceShape(getWindowWidth, getWindowHeight))
  }

  override def onGraphicRender(g: GdxGraphics): Unit = {
    g.clear()

    for (shape <- s) {
      shape.drawShape(g)
      shape.moveShape(g.getScreenWidth, g.getScreenHeight)
      shape.shiftShapeColour(frame)
      frame += 1
    }

    g.drawFPS()
    g.drawSchoolLogo()
  }
}

object DemoLines {
  def main(args: Array[String]): Unit = new DemoLines().launch()
}
